src/dmd/root/ctfloat.d

```diff
@@ -167,26 +167,43 @@ extern (C++) struct CTFloat
     }
 
     @system
-    static real_t parse(const(char)* literal, bool* isOutOfRange = null)
+    extern (D) private static real_t parse(alias parseFn)(const(char)* literal, bool* isOutOfRange)
```

Comment:
`isOutOfRange` should be `out bool`.

Reply:
That only complicates matters for the actual functions forwarding to this - they all take an optional pointer.

Reply:
Are there any that do this? Generally, I've been trying to move away from this old C style.

Reply:
Well, it's the usual solution for an optional `out` param. Parsing these literals occurs in 2 places here upstream - the lexer, which cares about any overflow, and hdrgen, which doesn't care about it because it's just used to determine whether the `%g` decimal string can be losslessly re-parsed to exactly the same value. In LDC, we have a bunch of additional calls, and IIRC, we don't care about the overflow there either.